main/persistent_storage/persistent_storage.c

```diff
@@ -18,7 +18,9 @@ esp_err_t persistent_storage_set_str(
 	return ESP_OK;
 }
 
-esp_err_t persistent_storage_set_wifi(const char* ssid, const char* password) {
+esp_err_t persistent_storage_set_connection_data(
+	connection_data_t connection_data
```

Comment:
Could you make this take in a pointer, rather than a base struct? Since right now it does a pretty expensive memory copy when it could just take in a pointer.